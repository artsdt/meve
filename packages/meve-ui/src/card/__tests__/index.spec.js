import Card from '..'
import { mount } from '@vue/test-utils'

test('test card pass description', () => {
  const wrapper = mount(Card, {
    propsData: {
      description: 'description',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test card pass string prop', () => {
  const wrapper = mount(Card, {
    propsData: {
      title: 'title',
      subtitle: 'subtitle',
      description: 'description',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test card slots', () => {
  const wrapper = mount(Card, {
    propsData: {
      title: 'title',
      subtitle: 'subtitle',
      description: 'description',
    },
    scopedSlots: {
      title: '<div>title</div>',
      subtitle: '<div>subtitle</div>',
      default: '<div>description</div>',
      extra: '<div>extra</div>',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})
