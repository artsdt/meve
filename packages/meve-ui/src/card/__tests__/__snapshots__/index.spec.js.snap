// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test card pass description 1`] = `
"<div description=\\"description\\" class=\\"m-card\\">
  <div class=\\"m-card__description\\">description</div>
</div>"
`;

exports[`test card pass string prop 1`] = `
"<div title=\\"title\\" subtitle=\\"subtitle\\" description=\\"description\\" class=\\"m-card\\">
  <div class=\\"m-card__title-wrapper\\">
    <div class=\\"m-card__title\\">title</div><small class=\\"m-card__subtitle\\">subtitle</small>
  </div>
  <div class=\\"m-card__description\\">description</div>
</div>"
`;

exports[`test card slots 1`] = `
"<div title=\\"title\\" subtitle=\\"subtitle\\" description=\\"description\\" class=\\"m-card\\">
  <div class=\\"m-card__title-wrapper\\">
    <div>title</div>
    <div>subtitle</div>
  </div>
  <div>description</div>
  <div>extra</div>
</div>"
`;
